src/sbl_filing_api/services/submission_processor.py

```diff
@@ -69,60 +75,67 @@ async def validate_and_update_submission(
             submission.state = SubmissionState.VALIDATION_IN_PROGRESS
             submission = await update_submission(session, submission)
 
-            df = pd.read_csv(BytesIO(content), dtype=str, na_filter=False)
-            submission.total_records = len(df)
+            file_path = generate_file_path(period_code, lei, submission.id)
+
+            final_phase = ValidationPhase.LOGICAL
+            all_findings = []
+            final_df = pl.DataFrame()
 
-            # Validate Phases
-            results = validate_phases(df, {"lei": lei}, max_errors=settings.max_validation_errors)
+            for validation_results in validate_batch_csv(
+                file_path, context={"lei": lei}, batch_size=50000, batch_count=1
+            ):
+                final_phase = validation_results.phase
+                all_findings.append(validation_results)
+
+            if all_findings:
+                final_df = pl.concat([v.findings for v in all_findings], how="diagonal")
 
-            submission.validation_results = build_validation_results(results)
+            submission.validation_results = build_validation_results(final_df, all_findings, final_phase)
 
-            if results.findings.empty:
+            if final_df.is_empty():
                 submission.state = SubmissionState.VALIDATION_SUCCESSFUL
             elif (
-                results.phase == ValidationPhase.SYNTACTICAL
+                final_phase == ValidationPhase.SYNTACTICAL
                 or submission.validation_results["logic_errors"]["total_count"] > 0
             ):
                 submission.state = SubmissionState.VALIDATION_WITH_ERRORS
             else:
                 submission.state = SubmissionState.VALIDATION_WITH_WARNINGS
 
-            submission_report = df_to_download(
-                results.findings,
-                warning_count=results.warning_counts.total_count,
-                error_count=results.error_counts.total_count,
-                max_errors=settings.max_validation_errors,
-            )
-            upload_to_storage(
-                period_code, lei, str(submission.id) + REPORT_QUALIFIER, submission_report.encode("utf-8")
-            )
+            report_path = generate_file_path(period_code, lei, f"{submission.id}_report")
+
+            df_to_download(final_df, report_path)
 
+            # upload_to_storage(
+            #    period_code, lei, str(submission.id) + REPORT_QUALIFIER, submission_report.encode("utf-8")
+            # )
             if not exec_check["continue"]:
                 log.warning(f"Submission {submission.id} is expired, will not be updating final state with results.")
                 return
 
             await update_submission(session, submission)
 
-        except RuntimeError:
-            log.exception("The file is malformed.")
+        except RuntimeError as re:
+            log.exception("The file is malformed.", re)
```

Comment:
this shouldn't be needed, log.exception contains `exc_info=True`, which extracts the exception from the stack.